src/e2e/specs/dashboard.spec.ts

```diff
@@ -269,38 +274,83 @@ test.describe(`${process.env.E2E_TEST_ENV} - Breaches Dashboard  - Payment`, ()
       },
     );
 
-    // verify subscription dialog elements
     await dashboardPage.subscribeButton.click();
-    await expect(dashboardPage.subscribeDialogCloseButton).toBeVisible();
-    await expect(dashboardPage.yearlyMonthlyTablist).toBeVisible();
-
-    // verify user purchase choices
-    await dashboardPage.yearlyTab.click();
-    await expect(
-      dashboardPage.subscribeDialogSelectYearlyPlanLink,
-    ).toBeVisible();
-
-    await dashboardPage.monthlyTab.click();
-    await expect(
-      dashboardPage.subscribeDialogSelectMonthlyPlanLink,
-    ).toBeVisible();
-
-    // Check monthly redirection
-    await dashboardPage.subscribeDialogSelectMonthlyPlanLink.click();
-    await purchasePage.subscriptionHeader.waitFor();
-    await expect(purchasePage.planDetails).toContainText("monthly");
-    await expect(purchasePage.subscriptionHeader).toBeVisible();
-
-    // Check yearly redirection
-    await page.goto(`${process.env.E2E_TEST_BASE_URL}`);
-    await dashboardPage.subscribeButton.waitFor();
-    await dashboardPage.subscribeButton.click();
-    await dashboardPage.subscribeDialogSelectYearlyPlanLink.click();
-    await purchasePage.subscriptionHeader.waitFor();
-    // Text contains invisible characters which need to be hardcoded in an assertion
-    await expect(purchasePage.planDetails).toContainText(
-      `${process.env.E2E_TEST_ENV === "prod" ? "yearly" : "every ⁨2⁩ months"}`,
+    await dashboardPage.verifyPremiumUpsellModalOptions();
+  });
+});
+
+test.describe(`${process.env.E2E_TEST_ENV} - Breaches Dashboard - Breaches Scan, Continuous Protection, Data Profile Actions`, () => {
+  test.use({ storageState: { cookies: [], origins: [] } });
+
+  test.beforeEach(async ({ landingPage, page, authPage, welcomePage }) => {
+    test.slow(
+      true,
+      "this test runs through the welcome scan flow, increasing timeout to address it",
     );
+
+    // speed up test by ignoring non necessary requests
+    await page.route(/(analytics)/, async (route) => {
+      await route.abort();
+    });
+
+    await landingPage.open();
+    await landingPage.goToSignIn();
+
+    const randomEmail = `_${Date.now()}@restmail.net`;
+    await authPage.signUp(randomEmail, page);
+
+    await page.waitForURL("**/user/welcome");
+    await welcomePage.goThroughFirstScan();
+    expect(page.url()).toContain("/user/dashboard");
```

Comment:
nit: we could also use `expect(page).toHaveURL(regexpOrString)`, but this works too.